packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -812,6 +849,61 @@ export class FluidDataStoreRuntime
 	}
 
 	public getAttachSummary(telemetryContext?: ITelemetryContext): ISummaryTreeWithStats {
+		const summaryBuilder = new SummaryTreeBuilder();
+		this.getAttachData((contextId: string, context: LocalChannelContextBase) => {
```

Comment:
I got really confused how this uses `this.getAttachData` which doesn't really return anything but puts sstuff in the summaryBuilder.
It would be lot cleaner and readable to just duplicate this code in the 2 functions.

Reply:
Good feedback.  I renamed it to make the interaction clearer.  It's kind of a visitor pattern.